src/sleep/sleep.c

```diff
@@ -78,7 +79,14 @@ static int write_hibernate_location_info(void) {
         if (r < 0)
                 return log_debug_errno(errno, "Unable to stat %s: %m", device);
 
-        // TODO check for btrfs and fail if offset is not provided; calculation will fail
+        r = btrfs_is_filesystem(fd);
+        if (r < 0)
+                return log_error_errno(r, "Error checking %s for Btrfs filesystem", device);
+
+        if (r)
+                return log_debug_errno(SYNTHETIC_ERRNO(EIO),
```

Comment:
not that it matters much, but I figure EOPNOTSUPP sounds more appropriate here than EIO

Reply:
Thanks for this suggestion. I reviewed `errno-base.h` and selected `EIO` which seemed like the best in that list but I see now that `errno.h` has a much more complete list. If you have any advice or a source I could review for how to select appropriate `errno`'s, I'd appreciate that.

Reply:
You can use `errno` (the binary).

Reply:
And apparently `man errno`... d'oh! :disappointed_relieved: